packages/data-point/lib/entity-types/entity-collection/factory.js

```diff
@@ -40,14 +40,11 @@ function createCompose (composeSpec) {
  */
 function create (spec, id) {
   validateModifiers(id, spec, modifierKeys.concat('compose'))
-  parseCompose.validateComposeModifiers(id, spec, modifierKeys)
 
   const entity = createBaseEntity(EntityCollection, spec, id)
-
-  const composeSpec = parseCompose.parse(spec, modifierKeys)
-  parseCompose.validateCompose(entity.id, composeSpec, modifierKeys)
-  if (composeSpec.length) {
-    entity.compose = createCompose(composeSpec)
+  const compose = parseCompose.parse(id, modifierKeys, spec)
```

Comment:
refactored `parseCompose` so that all the functionality is contained by the `parse` function (instead of needing to individually call `validateComposeModifiers`, `parse`, and `validateCompose`)